identify/identify.py

```diff
@@ -42,6 +44,8 @@ def tags_from_path(path: str) -> Set[str]:
         return {DIRECTORY}
     if os.path.islink(path):
         return {SYMLINK}
+    if stat.S_ISSOCK(os.stat(path).st_mode):
+        return {SOCKET}
```

Comment:
I wonder if we should do a single `os.stat` call and then decide based on that -- as it is right now this is going to do at least 4 stat calls to determine these tags

Reply:
Do you mean `isdir`, `islink`, and others do `os.stat` under the hood?

Reply:
yep, and lexists

Reply:
For `os.access` do you recommend hand-rolling something that supports both posix & windows or leaving it as-is? Also for some of the public functions that check if a file exists & other properties, should I remove those checks (because the caller checks them) to save on stat calls, or preserve the behavior because they're public?

Reply:
Went with erring towards leaving more code as-is but lmk if there's anything else straightforward that I missed